Source/Core/Core/AchievementManager.cpp

```diff
@@ -338,7 +338,15 @@ AchievementManager::RichPresence AchievementManager::GetRichPresence() const
   return m_rich_presence;
 }
 
-const AchievementManager::NamedBadgeMap& AchievementManager::GetChallengeIcons() const
+const bool AchievementManager::AreChallengesUpdated()
```

Comment:
I'm nitpicking here but this feels a bit strange to me, as the name implies a simple `bool` result but it's actually doing some work by resetting the value.

Thoughts on splitting this in two?

```
bool AchievementManager::AreChallengesUpdated() const
{
  return m_challenges_updated;
}

void AchievementManager::ResetChallengesUpdated()
{
  m_challenges_updated = false;
}
```

Reply:
I don't think I'd split them given that the second method is only ever supposed to be called from the context of the first method. Would CheckForChallengeUpdates be a better name perhaps?